Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -1749,47 +1753,64 @@ public class TimeSlider: UIControl {
         if isObserving {
             removeObservers()
         }
-        map?.addObserver(self, forKeyPath: #keyPath(AGSMap.operationalLayers), options: [.new, .old], context: &TimeSlider.KVOContext)
-        scene?.addObserver(self, forKeyPath: #keyPath(AGSScene.operationalLayers), options: [.new, .old], context: &TimeSlider.KVOContext)
-        geoView?.addObserver(self, forKeyPath: #keyPath(AGSGeoView.timeExtent), options: .new, context: &TimeSlider.KVOContext)
+        
+        // Observe operationalLayers of map
+        mapLayersObservation = map?.observe(\.operationalLayers, options: [.new, .old], changeHandler: { [weak self] (map, change) in
+            //
+            // Make sure self is around
+            guard let strongSelf = self else {
+                return
+            }
+            
+            // Handle the change in operationalLayers
+            strongSelf.handleOperationalLayers(change: change)
```

Comment:
Since you are only referencing `self` once, I would suggest removing the `guard` and just using `self?.handleOperationalLayers(change: change)`.

Same with the closure on lines `1770` and `2165`.

Reply:
fixed